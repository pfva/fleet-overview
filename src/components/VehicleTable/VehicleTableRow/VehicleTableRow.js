import VehicleTableCell from '../VehicleTableCell/VehicleTableCell';
import styles from './VehicleTableRow.module.css';

const VehicleTableRow = ({ vehicle }) => {
  const { id, name, driver, status, fuelType, equipments } = vehicle;

  return (
    <div className={styles.vehicleTableRow} data-testid="VehicleTableRow">
      <VehicleTableCell content={{ type: 'id', data: id, vehicleId: id }} />
      <VehicleTableCell content={{ type: 'name', data: name, vehicleId: id }} />
      <VehicleTableCell
        content={{ type: 'driver', data: driver, vehicleId: id }}
      />
      <VehicleTableCell
        content={{ type: 'status', data: status, vehicleId: id }}
      />
      <VehicleTableCell
        content={{ type: 'fuelType', data: fuelType, vehicleId: id }}
      />
      <VehicleTableCell
        content={{ type: 'equipments', data: equipments, vehicleId: id }}
      />
    </div>
  );
};

export default VehicleTableRow;
