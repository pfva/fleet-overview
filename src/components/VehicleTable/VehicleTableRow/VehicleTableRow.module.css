.vehicleTableRow {
  display: flex;
  background: var(--color-white);
  margin: var(--spacing-normal) 0;
  padding: var(--spacing-normal);
  justify-content: space-between;
}
