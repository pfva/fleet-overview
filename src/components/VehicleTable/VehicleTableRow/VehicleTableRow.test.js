import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import VehicleTableRow from './VehicleTableRow';

const mockedVehicle = {
  id: 'v3',
  name: 'GA33',
  driver: 'Squidward Tentacles',
  status: 'active',
  fuelType: 'CNG',
  equipments: [4], // Hook
};

describe('VehicleTableRow', () => {
  it('renders without crashing', () => {
    const tree = renderer
      .create(<VehicleTableRow vehicle={mockedVehicle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the id value', () => {
    render(<VehicleTableRow vehicle={mockedVehicle} />);
    const id = screen.getByText(mockedVehicle.id);
    expect(id).toBeInTheDocument();
  });

  it('renders the name value', () => {
    render(<VehicleTableRow vehicle={mockedVehicle} />);
    const name = screen.getByText(mockedVehicle.name);
    expect(name).toBeInTheDocument();
  });

  it('renders the driver value', () => {
    render(<VehicleTableRow vehicle={mockedVehicle} />);
    const driver = screen.getByText(mockedVehicle.driver);
    expect(driver).toBeInTheDocument();
  });

  it('renders the status value', () => {
    render(<VehicleTableRow vehicle={mockedVehicle} />);
    const status = screen.getByText(mockedVehicle.status);
    expect(status).toBeInTheDocument();
  });

  it('renders the fuelType value', () => {
    render(<VehicleTableRow vehicle={mockedVehicle} />);
    const fuelType = screen.getByText(mockedVehicle.fuelType);
    expect(fuelType).toBeInTheDocument();
  });

  it('renders the translated equipments value', () => {
    render(<VehicleTableRow vehicle={mockedVehicle} />);
    const equipments = screen.getByText(/Hook/i);
    expect(equipments).toBeInTheDocument();
  });
});
