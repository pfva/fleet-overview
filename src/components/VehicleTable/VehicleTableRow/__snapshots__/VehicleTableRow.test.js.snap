// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleTableRow renders without crashing 1`] = `
<div
  className="vehicleTableRow"
  data-testid="VehicleTableRow"
>
  <p
    className="vehicleTableCell"
    data-testid="VehicleTableCell-text"
    onClick={[Function]}
  >
    <span
      className="dataText"
    >
      v3
    </span>
  </p>
  <p
    className="vehicleTableCell"
    data-testid="VehicleTableCell-text"
    onClick={[Function]}
  >
    <span
      className="dataText"
    >
      GA33
    </span>
  </p>
  <p
    className="vehicleTableCell"
    data-testid="VehicleTableCell-text"
    onClick={[Function]}
  >
    <span
      className="dataText"
    >
      Squidward Tentacles
    </span>
  </p>
  <div
    className="vehicleTableCell iconWrapper"
    data-testid="VehicleStatus-on"
    onClick={[Function]}
  >
    <span
      className="dot"
      data-testid="Icon-active"
    />
    <p>
      <span
        className="dataText"
      >
        active
      </span>
    </p>
  </div>
  <p
    className="vehicleTableCell"
    data-testid="VehicleTableCell-text"
    onClick={[Function]}
  >
    <span
      className="dataText"
    >
      CNG
    </span>
  </p>
  <div
    className="vehicleEquipments"
    data-testid="VehicleEquipments-wrapper"
    onClick={[Function]}
  >
    <span
      className="dataText adjustHovered"
    >
      <p
        className="vehicleTableCell"
      >
        Hook
      </p>
    </span>
  </div>
</div>
`;
